src/presentation/utils/boltz.ts

```diff
@@ -0,0 +1,197 @@
+  const preimageHash = crypto.sha256(preimage).toString('hex');
+  return paymentHash === preimageHash;
+};
+
+// validates if reverse swap address derives from redeem script
+const reverseSwapAddressDerivesFromScript = (lockupAddress: string, redeemScript: string) => {
+  const addressScript = address.toOutputScript(lockupAddress);
+  const addressScriptASM = script.toASM(script.decompile(addressScript) || []);
+  const sha256 = crypto.sha256(Buffer.from(redeemScript, 'hex')).toString('hex');
+  const expectedAddressScriptASM = `OP_0 ${sha256}`; // P2SH
+  return addressScriptASM === expectedAddressScriptASM;
+};
+
+// validates if we can redeem with this redeem script
+const validReverseSwapReedemScript = (preimage: Buffer, pubKey: string, redeemScript: string) => {
+  const scriptAssembly = script
+    .toASM(script.decompile(Buffer.from(redeemScript, 'hex')) || [])
+    .split(' ');
+  const cltv = scriptAssembly[10];
+  const refundPubKey = scriptAssembly[13];
+  const expectedScript = [
+    'OP_SIZE',
+    '20',
+    'OP_EQUAL',
+    'OP_IF',
+    'OP_HASH160',
+    crypto.hash160(preimage).toString('hex'),
+    'OP_EQUALVERIFY',
+    pubKey,
+    'OP_ELSE',
+    'OP_DROP',
+    cltv,
+    'OP_NOP2',
+    'OP_DROP',
+    refundPubKey,
+    'OP_ENDIF',
+    'OP_CHECKSIG',
+  ];
+  return scriptAssembly.join() === expectedScript.join();
+};
+
+export const isValidReverseSubmarineSwap = (
+  invoice: string,
+  lockupAddress: string,
+  preimage: Buffer,
+  pubKey: string,
+  redeemScript: string
+): boolean => {
+  return (
+    correctPaymentHashInInvoice(invoice, preimage) &&
+    reverseSwapAddressDerivesFromScript(lockupAddress, redeemScript) &&
+    validReverseSwapReedemScript(preimage, pubKey, redeemScript)
+  );
+};
+
+export const getInvoiceTag = (invoice: string, tag: string): TagData => {
+  const decodedInvoice = bolt11.decode(invoice);
+  for (const { tagName, data } of decodedInvoice.tags) {
+    if (tagName === tag) return data;
+  }
+  return '';
+};
+
+export const getInvoiceValue = (invoice: string): number => {
+  const { satoshis, millisatoshis } = bolt11.decode(invoice);
+  if (satoshis) return fromSatoshi(satoshis);
+  if (millisatoshis) return fromSatoshi(Number(millisatoshis) / 1000);
+  return 0;
+};
+
+export const getInvoiceExpireDate = (invoice: string): number => {
+  const { timeExpireDate } = bolt11.decode(invoice);
+  return timeExpireDate ? timeExpireDate * 1000 : 0; // milliseconds
+};
+
+export const getClaimTransaction = async (
+  account: MnemonicAccount,
+  addr: AddressInterface,
+  explorerURL: string,
+  network: NetworkString,
+  password: string,
+  preimage: Buffer,
+  redeemScript: string,
+  utxos: Outpoint[]
+): Promise<Transaction> => {
+  // utxo has arrived, prepare claim transaction
+  const [utxo] = utxos;
+  const hex = await fetchTxHex(utxo.txid, explorerURL);
+  const prevout = Transaction.fromHex(hex).outs[utxo.vout];
+
+  const tx = new Psbt({ network: (networks as any)[network] });
```

Comment:
You can use [getNetwork](https://github.com/vulpemventures/ldk/blob/7261fcdcb0116ec207780c3dde12d96bcef13489/src/utils.ts#L185-L187) from LDK to avoid `as any`

Reply:
Fixed by 305da4f